base62.go

```diff
@@ -1,17 +1,17 @@
 package ksuid
 
-import "bytes"
+import "strings"
 
 // lexographic ordering (based on Unicode table) is 0-9A-Za-z
-var base62Characters = []byte("0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz")
+var base62Characters = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz"
 
-func base2base(in []byte, inBase int, outBase int) []byte {
-	res := []byte{}
-	bs := in[:]
+func appendBase2Base(dst []byte, src []byte, inBase int, outBase int) []byte {
```

Comment:
Yes, this is some `append` magic. A slice tracks the lengths and the capacity of the backend array it _points_ to, if there's room it'll use it and won't reallocate memory.

More details here => https://blog.golang.org/slices